tests/test_common.py

```diff
@@ -19,6 +22,99 @@
 from tftbase import TestType  # noqa: E402
 
 
+def test_str_to_bool() -> None:
+    assert common.str_to_bool(True) is True
+    assert common.str_to_bool(False) is False
+
+    ERR = object()
+    DEF = object()
+
+    # falsy
+    assert common.str_to_bool("0") is False
+    assert common.str_to_bool("n") is False
+    assert common.str_to_bool("no") is False
+    assert common.str_to_bool("false") is False
+
+    # truthy
+    assert common.str_to_bool("1") is True
+    assert common.str_to_bool("y") is True
+    assert common.str_to_bool("yes") is True
+    assert common.str_to_bool("true") is True
+
+    assert common.str_to_bool("bogus", None) is None
```

Comment:
I don't understand this test case:
Should it be assert common.str_to_bool("bogus") is None. Why do we need to pass additional arguments to str_to_bool?

Reply:
if you don't pass a second argument, then `str_to_bool()` will raise an exception on unrecognized strings. That's a choice, by default it's strict in what it accepts.

This tests sets the `on_error` argument to `None`, which enables that any invalid argument is returned as the on-error value.

There is also a test
```python
    with pytest.raises(ValueError):
        common.str_to_bool("bogus")
```
which covers the case you mention.

Reply:
Can we add that test instead? I think it is more useful instead of using on_error.

Reply:
Which test? I think that `test_str_to_bool()` covers *a lot* of cases (some even cases redundantly, which I don't mind).

If you think a case is missing, I don't see it.

Reply:
Ah sorry, I had to look further down. Yes I see that you added that case as well. Thanks! This is ok for me.